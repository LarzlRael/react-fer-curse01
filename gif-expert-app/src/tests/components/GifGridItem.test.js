import { shallow } from "enzyme"
import GifGridItem from "../../components/GifGridItem";



describe('Testing in <GifGridItem />', () => {

    const title = 'Un titulo';
    const url = 'https://localhost:8080';

    const wrapper = shallow(
        <GifGridItem
            title={title}
            url={url}
        />);

    test('should show correctly the component with props', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('It must have a paragraph with a title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
        expect(wrapper).toMatchSnapshot();
    })

    test('It must have the image by the url and the alt by props', () => {
        const img = wrapper.find('img');
        // find the tag properties
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('It must have animate_fadeIn', () => {

        const div = wrapper.find('div');
        // get the className 
        const className = div.prop('className');

        // if class name it must have the class name is true o false
        expect(className.includes('animate__fadeIn')).toBe(true);
    })
})
