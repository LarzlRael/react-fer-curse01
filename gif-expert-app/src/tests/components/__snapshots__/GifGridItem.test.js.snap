// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <GifGridItem /> It must have a paragraph with a title 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost:8080"
  />
  <p>
    Un titulo
  </p>
</div>
`;

exports[`Testing in <GifGridItem /> should show correctly the component with props 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost:8080"
  />
  <p>
    Un titulo
  </p>
</div>
`;
